tests/managed_array_tests.cpp

```diff
@@ -369,6 +412,28 @@ CUDA_TEST(ManagedArray, IncrementDecrementFromHostOnDeviceUM) {
 
   array.free();
 }
+
+CUDA_TEST(ManagedArray, PickandSetSliceDeviceToDeviceUM) {
+  chai::ManagedArray<int> array(10, chai::UM);
+  chai::ManagedArray<int> sl1 = array.slice(0,5);
+  chai::ManagedArray<int> sl2 = array.slice(5,5);
+
+  forall(cuda(), 0, 10, [=] __device__ (int i) {
+      array[i] = i;
+  });
+
+  forall(cuda(), 0, 5, [=] __device__ (int i) {
```

Comment:
Is there a way to write the unit test to check that only one device to host transfer occurs before the sequential loop in this test?